src/AutoRest.CSharp/LowLevel/Output/OperationMethodChainBuilder.cs

```diff
@@ -116,6 +116,59 @@ private bool IsConvenienceMethodMeaningful()
                 || !_returnType.Convenience.Equals(_returnType.Protocol);
         }
 
+        private bool IsLowConfident()
+        {
+            var visitedModels = new HashSet<InputModelType>();
+            // check response
+            var response = Operation.Responses.Where(r => !r.IsErrorResponse).FirstOrDefault();
+            if (response?.BodyType is { } responseBodyType && IsLowConfident(responseBodyType, visitedModels))
+            {
+                return true;
+            }
+            // check parameters
+            if (Operation.Parameters.Where(p => !p.IsContentType).Any(parameter => IsLowConfident(parameter.Type, visitedModels)))
+            {
+                return true;
+            }
+
+            return false;
+        }
+
+        private static bool IsLowConfident(InputType type, HashSet<InputModelType> visitedModels)
+        {
+            var result = type switch
+            {
+                InputLiteralType literalType => IsLowConfident(literalType.LiteralValueType, visitedModels),
+                InputListType listType => IsLowConfident(listType.ElementType, visitedModels),
+                InputDictionaryType dictType => IsLowConfident(dictType.KeyType, visitedModels) || IsLowConfident(dictType.ValueType, visitedModels),
+                InputModelType modelType => IsModelLowConfident(modelType, visitedModels),
+                _ => type.IsLowConfident
+            };
+            return result;
+        }
+
+        private static bool IsModelLowConfident(InputModelType modelType, HashSet<InputModelType> visitedModels)
+        {
+            if (visitedModels.Contains(modelType)) // if this has been checked or is being checked, we just return false and let others to decide this
```

Comment:
```C#
class NestedModel
{
    public AUnion{set; get;}
}
class Model
{
    public NestedModel {get; set;}
}
```

After NestedModel checked, the Model will return false, which acutually should return true.

Reply:
make sense. let me see how to resolve this

Reply:
this should be resolved now.